exec/java-exec/src/main/java/org/apache/drill/exec/util/UserDelegationUtil.java

```diff
@@ -0,0 +1,147 @@
+
+  private static final ObjectMapper delegationDefinitionsMapper = new ObjectMapper();
+
+  static {
+    delegationDefinitionsMapper.configure(JsonGenerator.Feature.QUOTE_FIELD_NAMES, false);
+    delegationDefinitionsMapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
+  }
+
+  private static class DelegationDefinition {
+    public UserGroupDefinition delegates = new UserGroupDefinition();
+    public UserGroupDefinition delegators = new UserGroupDefinition();
+  }
+
+  private static class UserGroupDefinition {
+    public Set<String> users = Sets.newHashSet();
+    public Set<String> groups = Sets.newHashSet();
+  }
+
+  /**
+   * Deserialize delegation definitions string to a list of delegation definition objects.
+   *
+   * @param delegationDefinitions delegation definitions as a sting
+   * @return delegation definitions as a list of objects
+   * @throws IOException
+   */
+  public static List<DelegationDefinition> deserializeDelegationDefinitions(final String delegationDefinitions)
+      throws IOException {
+    return delegationDefinitionsMapper.readValue(delegationDefinitions,
+        new TypeReference<List<DelegationDefinition>>() {});
+  }
+
+  /**
+   * Validator for delegation definitions.
+   */
+  public static class DelegationDefinitionsValidator extends TypeValidators.AdminOptionValidator {
+
+    public DelegationDefinitionsValidator(String name, String def) {
+      super(name, def);
+    }
+
+    @Override
+    public void validate(OptionValue v) {
+      super.validate(v);
+
+      final List<DelegationDefinition> definitions;
+      try {
+        definitions = deserializeDelegationDefinitions(v.string_val);
+      } catch (final IOException e) {
+        throw UserException.validationError()
+            .message("Invalid delegation definition.\nDetails: %s", e.getMessage())
+            .build(logger);
+      }
+
+      for (final DelegationDefinition definition : definitions) {
+        if (definition.delegates.users.contains(STAR) ||
+            definition.delegates.groups.contains(STAR)) {
+          throw UserException.validationError()
+              .message("No wildcard delegates allowed.")
+              .build(logger);
+        }
+      }
+    }
+  }
+
+  /**
+   * Check if the given delegate is authorized to delegate for the delegator based on the delegation definitions.
+   *
+   * @param delegateName          delegate name
+   * @param delegatorName         delegator name
+   * @param delegationDefinitions valid delegation definitions
+   * @return true iff delegate is authorized to delegate for the delegator
+   */
+  public static boolean hasDelegationPrivileges(final String delegateName, final String delegatorName,
+                                                final String delegationDefinitions) throws IOException {
+    final List<DelegationDefinition> definitions;
+    try {
+      definitions = deserializeDelegationDefinitions(delegationDefinitions);
```

Comment:
Where would this representation (list of DelegationDefinition objects) live? UserServer?